client/my-sites/site-settings/index.js

```diff
@@ -63,6 +63,14 @@ export default function() {
 			settingsController.setScroll,
 			controller.disconnectSite
 		);
+
+		page(
+			'/settings/disconnect-site/:site_id/confirm',
```

Comment:
We've been trying to keep the site fragment at the end of routes for various reasons (consistency, and for helping some utilities do their job, like `getSiteFragment`), so I'd alter this one to:

```
'/settings/disconnect-site/confirm/:site_id',
```

To reproduce an error with the `:site_id` not being at the end, while in the disconnect confirm page, switch your site. You'll be redirected to `settings/disconnect-site/confirm/:site` (notice where the site slug is).

Reply:
Ah, good point. I modelled this after the wizard in https://github.com/Automattic/wp-calypso/blob/30d4eacd8a0b88cf2f8eaea766b85db418900506/client/extensions/wp-job-manager/index.js#L44, but yeah, the reason/confirm fragment should proably be next to `disconnect-site`.